### PR TITLE
fix: add notification center registry

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
Summary
-------
Create a private registry of NotificationCenters for internal use.
- Remove dependence on user supplied NotificationCenter.
- Resolve issue of notifications being registered with one NotificationCenter and sent to a different one.

Test plan
---------
added tests to
- test_notification_center_registry.py
- test_optimizely.py
- test_optimizely_config.py
- test_optimizely_factory.py

Ticket
------

[FSSDK-8773](https://jira.sso.episerver.net/browse/FSSDK-8773)

### CODE REVIEW

File: optimizely/optimizely.py [hunk 1]
```diff
@@ -174,8 +163,8 @@ def __init__(
             else:
                 self.config_manager = StaticConfigManager(**config_manager_options)
 
-        if not self.sdk_settings.odp_disabled:
-            self._update_odp_config_on_datafile_update()
+        self.odp_manager: OdpManager
+        self.setup_odp(sdk_key or self.config_manager.get_sdk_key())
```
Comment (jaeopt) [inline optimizely/optimizely.py:167]: I see 2 changes related to "sdk_key" in here and in config_manager, both optional.
If none of them required, then we may have a case sdk_key is null.
As noted in my comment in config_manager, we may not need get_sdk_key in config_manager. 
We can consider "sdk_key" in optimizely from optional to required while removing the changes in config_manager.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Changes look good.  I see a case when sdk_key is not provided. See my comments.

File: tests/test_notification_center_registry.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Copyright 2019, Optimizely
```
Comment (Mat001) [inline tests/test_notification_center_registry.py:1]: 2023?

**Review by Mat001 — APPROVED** [review]
Looks ok. One typo

**Review by jaeopt — APPROVED** [review]
LGTM

**Comment by jaeopt** [comment]
@andrewleap-optimizely can we keep a note about this breaking change. We should release a major version bump with this note.

**Pull request merged**